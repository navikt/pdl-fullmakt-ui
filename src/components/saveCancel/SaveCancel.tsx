import * as React from 'react';
import { I18n } from 'react-i18nify';
import { MouseEventHandler } from 'react';

interface SaveCancelProps {
  id?: string;
  saveLabel?: string;
  saveDisabled?: boolean;
  onClickSave?: MouseEventHandler<any>;
  cancelLabel?: string;
  cancelDisabled?: boolean;
  onClickCancel?: MouseEventHandler<any>;
  translate?: Function;
}

const SaveCancel = (props: SaveCancelProps) => (
  <div className={'row'}>
    <div
      className="col-xs-6 col-sm-5 col-md-2 col-mdOffset-d"
      style={{ border: 'none', textAlign: 'center' }}
    >
      <button
        id={`Save_${props.id || 1}`}
        onClick={props.onClickSave}
        className="btn btn-outline-primary"
        type="button"
        disabled={props.saveDisabled}
      >
        {props.saveLabel || I18n.t('fullmakt.words.save')}
      </button>
    </div>
    <div
      className="col-xs-6 col-sm-5 col-md-2 "
      style={{ border: 'none', textAlign: 'center' }}
    >
      <button
        id={`Cancel_${props.id || 1}`}
        onClick={props.onClickCancel}
        className="btn btn-link"
        type="button"
        disabled={props.cancelDisabled}
      >
        {props.cancelLabel || I18n.t('fullmakt.words.cancel')}
      </button>
    </div>
  </div>
);

SaveCancel.displayName = 'SaveCancel';

export default SaveCancel;
