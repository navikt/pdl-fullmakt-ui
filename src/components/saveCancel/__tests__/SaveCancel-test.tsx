import * as React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import SaveCancel from '../SaveCancel';
import * as sinon from 'sinon';

describe('SaveCancel snapshot', () => {
  it('should match snapshot', () => {
    const onClickSave = sinon.spy();

    const component = (
      <SaveCancel
        id="2"
        saveLabel="Hii"
        onClickSave={onClickSave}
        cancelDisabled={true}
      />
    );

    const wrapper = shallow(component);
    expect(wrapper.find('#Save_2').hasClass('disabled')).toEqual(false);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should react on clickCancel', () => {
    const onClickSave = sinon.spy();
    const onClickCancel = sinon.spy();
    const wrapper = shallow(
      <SaveCancel onClickCancel={onClickCancel} onClickSave={onClickSave} />
    );
    wrapper.find('#Cancel_1').simulate('click');
    expect(onClickSave.calledOnce).toBe(false);
    expect(onClickCancel.calledOnce).toBe(true);
  });

  it('should react on clickSave', () => {
    const onClickSave = sinon.spy();
    const onClickCancel = sinon.spy();
    const wrapper = shallow(
      <SaveCancel onClickCancel={onClickCancel} onClickSave={onClickSave} />
    );
    wrapper.find('#Save_1').simulate('click');
    expect(onClickSave.calledOnce).toBe(true);
    expect(onClickCancel.calledOnce).toBe(false);
  });
});
