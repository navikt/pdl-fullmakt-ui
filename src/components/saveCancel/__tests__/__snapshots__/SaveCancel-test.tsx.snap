// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveCancel snapshot should match snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-xs-6 col-sm-5 col-md-2 col-mdOffset-d"
    style={
      Object {
        "border": "none",
        "textAlign": "center",
      }
    }
  >
    <button
      className="btn btn-outline-primary"
      id="Save_2"
      onClick={[Function]}
      type="button"
    >
      Hii
    </button>
  </div>
  <div
    className="col-xs-6 col-sm-5 col-md-2 "
    style={
      Object {
        "border": "none",
        "textAlign": "center",
      }
    }
  >
    <button
      className="btn btn-link"
      disabled={true}
      id="Cancel_2"
      type="button"
    >
      Avbryt
    </button>
  </div>
</div>
`;
